components/expenses/ExpenseForm.js

```diff
@@ -335,17 +373,19 @@ const ExpenseForm = ({ onSubmit, collective, expense, payoutProfiles, autoFocusT
           setValidate(true);
           formik.setErrors(errors);
         } else {
+          formPersister.clearValues();
```

Comment:
**[:no_entry_sign: **Blocker**]** If `onSubmit` fails, this will clear the values. We should only clear them if the submit is successful. Because this component is used in both `create-expense` and `expense` (to edit), I suggest to move the formPersister in the `create-expense` page

---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>

Reply:
I will make the change. I'm assuming we will just pass `formPersister` as a prop to `ExpenseForm` in `create-expense` page.

Reply:
Yep this is a good approach. Also keep in mind that there's no reason to save the values in for the edit form, so `formPersister` must be an optional prop.